File: fsharp/hamming/Hamming.fs
module Hamming

let distance (strand1: string) (strand2: string) =
    if strand1.Length <> strand2.Length then
        None
    else
        Seq.zip strand1 strand2
        |> Seq.sumBy (fun (x, y) -> if x <> y then 1 else 0)
        |> Some

File: python/hamming/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    """
    Calculate the Hamming distance between two DNA strands.
    """
    try:
        return sum(a != b for a, b in zip(strand_a, strand_b, strict=True))
    except ValueError:
        raise ValueError("Strands must be of equal length.")
